core/src/main/java/org/mule/config/pool/MuleDiscardPolicy.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.pool;
+
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+
+public class MuleDiscardPolicy implements RejectedExecutionHandler
```

Comment:
Class level javadoc. How does this differ to DiscardOldestPolicy? Can it replace it? If we don't replace it then can we at least extend it?

Reply:
I've extended DiscardPolicy though in this case no logic can be reused (though it is correct extend that class).